src/angular-app/bellows/core/offline/editor-data.service.ts

```diff
@@ -327,66 +329,69 @@ export class EditorDataService {
+    const sensesField = config.entry.fields.senses as LexConfigFieldList;
     if (fieldKey in config.entry.fields && fieldKey in entry) {
       field = config.entry.fields[fieldKey];
       dataNode = entry[fieldKey];
-    } else if (fieldKey in config.entry.fields.senses.fields && angular.isDefined(entry.senses) &&
+    } else if (fieldKey in sensesField.fields && angular.isDefined(entry.senses) &&
       entry.senses.length > 0 && fieldKey in entry.senses[0]
     ) {
-      field = config.entry.fields.senses.fields[fieldKey];
+      field = sensesField.fields[fieldKey];
       dataNode = entry.senses[0][fieldKey];
     }
 
     if (field || isSpecialMultitext) {
-      if (isSpecialMultitext || field.type === 'multitext') {
-        // special case for lexeme form / citation form. Use citation form if available, fall back to lexeme form
-        if (fieldKey === 'lexeme' || fieldKey === 'citationForm') {
-          if (config.entry.fields.citationForm) {
-            const citationFormInputSystems = config.entry.fields.citationForm.inputSystems;
-            if (entry.citationForm && citationFormInputSystems.length > 0 &&
-              citationFormInputSystems[0] in entry.citationForm
-            ) {
-              sortableValue = entry.citationForm[citationFormInputSystems[0]].value;
-            }
-          }
-
-          if (!sortableValue) {
-            const lexemeInputSystems = config.entry.fields.lexeme.inputSystems;
-            if (entry.lexeme && lexemeInputSystems.length > 0 && lexemeInputSystems[0] in entry.lexeme) {
-              sortableValue = entry.lexeme[lexemeInputSystems[0]].value;
-            }
+      // special case for lexeme form / citation form. Use citation form if available, fall back to lexeme form
+      if (isSpecialMultitext) {
+        if ('citationForm' in config.entry.fields) {
+          const citationFormField = config.entry.fields.citationForm as LexConfigMultiText;
+          const citationFormInputSystems = citationFormField.inputSystems;
+          if (entry.citationForm && citationFormInputSystems.length > 0 &&
+            citationFormInputSystems[0] in entry.citationForm
+          ) {
+            sortableValue = entry.citationForm[citationFormInputSystems[0]].value;
           }
+        }
 
-        // regular multi-text field
-        } else {
-          if (field.inputSystems.length > 0 && field.inputSystems[0] in dataNode) {
-            sortableValue = dataNode[field.inputSystems[0]].value;
+        if (!sortableValue) {
+          const lexemeField = config.entry.fields.lexeme as LexConfigMultiText;
+          // don't sort by the first lexeme form if it is an "-audio" writing system
+          const inputSystem = lexemeField.inputSystems.find((system) => !system.includes('-audio'));
+          if (entry.lexeme && inputSystem && inputSystem in entry.lexeme) {
+            sortableValue = entry.lexeme[inputSystem].value;
```

Comment:
This is the core of the bugfix.

It does assume that citation form will never be an audio writing system (this is possible in LF, not sure if it is in FLEx) since this code only executes if citation form is empty.

Reply:
Good point. I pushed another commit to cover the citation form too.